pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -91,17 +92,33 @@ func (a *v1beta1PolicyApplier) setAuthnFilterForPeerAuthn(proxyType model.NodeTy
 	p := config.Policy
 	p.Peers = []*authn_alpha.PeerAuthenticationMethod{}
 
-	effectiveMTLSMode := a.getMutualTLSModeForPort(port)
-	if effectiveMTLSMode == model.MTLSPermissive || effectiveMTLSMode == model.MTLSStrict {
-		mode := authn_alpha.MutualTls_PERMISSIVE
-		if effectiveMTLSMode == model.MTLSStrict {
-			mode = authn_alpha.MutualTls_STRICT
+	if proxyType == model.SidecarProxy {
```

Comment:
I think this can be simplified a little bit by setting the `effectiveMTLSMode`, then you don't need to duplicate the common code for setting it in the peers.
```go
	var effectiveMTLSMode model.MutualTLSMode
	if proxyType == model.SidecarProxy {
		effectiveMTLSMode = a.getMutualTLSModeForPort(port)
	} else {
		// this is for gateway with a server whose TLS mode is ISTIO_MUTUAL
		// this is effectively the same as strict mode. We dont really
		// care about permissive or strict here. We simply need to validate that the peer cert is
		// a proper spiffe cert so that authz policies can use source principal based validations here.
		effectiveMTLSMode = model.MTLSStrict
	}
	if effectiveMTLSMode == model.MTLSPermissive || effectiveMTLSMode == model.MTLSStrict {
		mode := authn_alpha.MutualTls_PERMISSIVE
		if effectiveMTLSMode == model.MTLSStrict {
			mode = authn_alpha.MutualTls_STRICT
		}
		p.Peers = []*authn_alpha.PeerAuthenticationMethod{
			{
				Params: &authn_alpha.PeerAuthenticationMethod_Mtls{
					Mtls: &authn_alpha.MutualTls{
						Mode: mode,
					},
				},
			},
		}
	}
```